packages/trader/src/App/Components/Elements/Modals/ServicesErrorModal/__tests__/services-error-modal.spec.tsx

```diff
@@ -0,0 +1,112 @@
+            type: '',
+        };
+        const { container } = render(
+            <ServicesErrorModal
+                services_error={services_error_mock}
+                is_logged_in
+                is_virtual
+                is_visible
+                onConfirm={onConfirm}
+            />
+        );
+        expect(container).toBeEmptyDOMElement();
+    });
+
+    it('AuthorizationRequiredModal should render when code is AuthorizationRequired', () => {
+        const services_error_mock = {
+            code: 'AuthorizationRequired',
+            message: 'AuthorizationRequired',
+        };
+        const authorization_props = {
+            is_logged_in: true,
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...authorization_props} />);
+        expect(screen.getByText('AuthorizationRequiredModal')).toBeInTheDocument();
+    });
+
+    it('InsufficientBalanceModal should render when code is InsufficientBalance', () => {
+        const services_error_mock = {
+            code: 'InsufficientBalance',
+            message: 'InsufficientBalance',
+        };
+        const insufficient_balance_props = {
+            is_virtual: true,
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...insufficient_balance_props} />);
+        expect(screen.getByText('InsufficientBalanceModal')).toBeInTheDocument();
+    });
+
+    it('CompanyWideLimitExceededModal should render when code is CompanyWideLimitExceeded', () => {
+        const services_error_mock = {
+            code: 'CompanyWideLimitExceeded',
+            message: 'CompanyWideLimitExceeded',
+        };
+        const limit_exceeded_props = {
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...limit_exceeded_props} />);
+        expect(screen.getByText('CompanyWideLimitExceededModal')).toBeInTheDocument();
+    });
+
+    it('AccountVerificationRequiredModal should render when code is PleaseAuthenticate', () => {
+        const services_error_mock = {
+            code: 'PleaseAuthenticate',
+            message: 'PleaseAuthenticate',
+        };
+        const account_verification_required_props = {
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...account_verification_required_props} />);
+        expect(screen.getByText('AccountVerificationRequiredModal')).toBeInTheDocument();
+    });
+
+    it('Default case should render when code is not specified in switch case', () => {
+        const services_error_mock = {
+            code: 'Default Error',
+            message: 'Default Error',
+            title: 'Default Error',
+        };
+        const default_case_props = {
```

Comment:
You can create this variable under the describe and use it in every test case as 
`is_visible: true,
onConfirm: jest.fn(),`
are used in every component